### PR TITLE
Dynamic styling

### PR DESCRIPTION
**Pull request by DanyFS**
I decided to do it using inline style because I tried with import and templating, but I wasn’t able to inject it correctly… After loosing 30minutes trying to, I decided to use inline style. 

Drift did it with inline style, I know you would prefer to do it using an external theme.css with template...

### CODE REVIEW

File: webpack.js [hunk 1]
```diff
@@ -8,10 +8,14 @@ var nodeConfig = {
   output: {
     path: './bin',
     filename: 'node.bundle.js',
-    libraryTarget: 'commonjs2'
+    libraryTarget: 'commonjs2',
+    publicPath: __dirname
   },
-  externals: [nodeExternals(), 'botpress'],
+  externals: [nodeExternals()],
```
Comment (slvnperron) [inline webpack.js]: Why did you remove botpress from externals?

Comment (slvnperron) [inline webpack.js]: Please revert the webpack changes, I suspect this was an attempt from you to make it work server-side?

File: src/views/web/index.jsx [hunk 2]
```diff
@@ -1,7 +1,8 @@
 import React from 'react'
 import ReactDOM from 'react-dom'
 import classnames from 'classnames'
-import {Emoji} from 'emoji-mart'
+import axios from 'axios'
```
Comment (slvnperron) [inline src/views/web/index.jsx]: Don't include axios, botpress already inject axios to the component

Use this.props.bp.axios instead

File: src/index.js [hunk 3]
```diff
@@ -110,9 +112,43 @@ const startNewSession = (bp, socketId) => {
...
 }
 
-module.exports = {
+const createConfigFile = (bp) => {
+  const name = 'botpress-web.config.yml'
+  const file = path.join(bp.projectLocation, name)
+
+  if (!fs.existsSync(file)) {
+    
+    const template = fs.readFileSync(path.resolve(__dirname, '..' ,name))
```
Comment (slvnperron) [inline src/index.js]: Don't do this, just include the template in the bundle at bundling time with webpack bu using the raw-loader (see how we do it for inject.js and inject.css)

Comment (slvnperron) [inline src/index.js]: So you won't need all the weird webpack modifications you did

File: src/index.js [hunk 4]
```diff
@@ -110,9 +112,43 @@ const startNewSession = (bp, socketId) => {
   return getOrCreateUser(bp, socketId)
 }
 
-module.exports = {
+const createConfigFile = (bp) => {
```
Comment (slvnperron) [inline src/index.js]: We should have a default behavior in botpress for creating modules's default config files automatically? Seems like this method will be duplicated in every modules

File: webpack.js (hunk 1 above)
Comment (DanyFS) [inline webpack.js]: I work on the `webpack` file because I want to have access to `__dirname` as you can see below. It's the only reason, botpress needs to stay there..

File: src/views/web/index.jsx (hunk 2 above)
Comment (DanyFS) [inline src/views/web/index.jsx]: Fixed!

File: src/index.js (hunk 3 above)
Comment (DanyFS) [inline src/index.js]: Okay okay, I understand, but I'm not sure to understand because the file can't exist. 

What I try to do here is to create a default config (`botpress-web.config.yml`) if it doesn't exist, but I don't need to use it. `bp` manage the reading of the configuration file for us..

File: webpack.js (hunk 1 above)
Comment (slvnperron) [inline webpack.js]: botpress needs to stay there otherwise it won't compile, it will attempt to bundle botpress inside botpress-web

File: src/index.js [hunk 5]
```diff
@@ -56,6 +56,7 @@ module.exports = {
 
     createConfigFile(bp) //Create a config file
     umm(bp) // Initialize UMM
+    return umm(bp)
```
Comment (slvnperron) [inline src/index.js]: Duplicated line

File: src/views/web/index.jsx (hunk 2 above)
Comment (slvnperron) [inline src/views/web/index.jsx]: I don't see the change?!

File: src/index.js (hunk 3 above)
Comment (slvnperron) [inline src/index.js]: The reason you are playing with webpack is because you need `__dirname` to read the template file. What I am suggesting here is that instead of reading the template from the disk you instead bundle it with `raw-loader` like we did for `inject.js`. 

The file creation of `botpress-web.config.yml` is fine and you don't need to alter webpack.js for that, projectLocation will work just fine

File: webpack.js (hunk 1 above)
Comment (DanyFS) [inline webpack.js]: Fixed! Thanks for the explication, I add it to the externals.

File: src/views/web/index.jsx (hunk 2 above)
Comment (DanyFS) [inline src/views/web/index.jsx]: Fixed! I don't know why it didn't commit on my last commit, it was okay on my side. Strange!

File: src/index.js (hunk 3 above)
Comment (DanyFS) [inline src/index.js]: Thanks for the explication and the reasons of the modification again. I really like to have feedback, I learn a lot of stuff quickly by doing PR.

File: src/index.js (hunk 5 above)
Comment (DanyFS) [inline src/index.js]: Removed, strange again... On my side, it's not duplicated

File: src/index.js (hunk 4 above)
Comment (DanyFS) [inline src/index.js]: Agree!

(A) every modules should have an config file

OR

(B) at least, we should have a generic method on bp to do it, just by passing the name of the module?

Which solution do you prefer? I prefer B because some modules won't have any config..

**Comment by DanyFS** [comment]
I made the changes asked.

The only thing we need to figured out is how we want to implement the creation of **.config.yml** files. (*see comment in the relevant section*)

**Comment by slvnperron** [comment]
@DanyFS Thanks! Looks all good now. Let's push back the config file creation to an other time
